Gems/ScriptCanvas/Code/Include/ScriptCanvas/Debugger/Debugger.h

```diff
@@ -129,14 +129,14 @@ namespace ScriptCanvas
                 else if (m_state == SCDebugState_Attached)
                 {
                     const Signal& asSignal(nodeSignal);
-                    Breakpoint breakpoint(asSignal);
+                    Breakpoint breakpoint(asSignal.m_endpoint.GetNodeId());
 
                     if (m_breakpoints.find(breakpoint) != m_breakpoints.end())
                     {
                         SCRIPT_CANVAS_DEBUGGER_TRACE_SERVER("Hit breakpoint: %s", nodeSignal.ToString().data());
                         if (remoteToolsInterface)
                         {
-                            remoteToolsInterface->SendRemoteToolsMessage(m_client.m_info, Message::BreakpointHit(nodeSignal));
+                            remoteToolsInterface->SendRemoteToolsMessage(m_client.m_info, Message::BreakpointHit(breakpoint));
```

Comment:
This is the important area, where the client is checking that the signal sent from scriptcanvas lua during execution matches one of its breakpoint, if so, the Interact() will enter a while(true) loop only updating the remote tools messages (logic is similar in lua debugger, works like a charm there because lua editor is already a separate executable from the client)